blog/2023-08-01-tolgee-with-next-13.md

```diff
@@ -0,0 +1,347 @@
+
+import { getTranslate } from 'tolgee/server';
+import { Todos } from './Todos';
+import Link from 'next-intl/link';
+
+import { Navbar } from 'components/Navbar';
+
+export default async function IndexPage() {
+  // because this is server component, use `getTranslate`
+  // not useTranslate from '@tolgee/react'
+  const t = await getTranslate();
+  return (
+    <main>
+      <h1>{t('page-example-title')}</h1>
+    </main>
+  );
+}
+```
+
+If everything is set up correctly, the 'page-example-title' should be `alt + click`able. Make sure you've defined your project credentials in the `.env.development.local` file.
+
+For client components, you can use the regular `React` integration:
+
+```tsx
+'use client';
+
+import { useTranslate } from '@tolgee/react';
+
+export const ExampleClientComponent = () => {
+  const { t } = useTranslate();
+
+  return (
+    <section>
+      <span>{t('example-key-in-client-component')}</span>
+    </section>
+  );
+};
+```
+
+### Switching Languages
+
+For switching languages use the following code:
+
+```tsx
+'use client';
+
+import React, { ChangeEvent, useTransition } from 'react';
+import { usePathname, useRouter } from 'next-intl/client';
+import { useTolgee } from '@tolgee/react';
+
+export const LangSelector: React.FC = () => {
+  const tolgee = useTolgee(['language']);
+  const locale = tolgee.getLanguage();
+  const router = useRouter();
+  const pathname = usePathname();
+  const [isPending, startTransition] = useTransition();
+
+  function onSelectChange(event: ChangeEvent<HTMLSelectElement>) {
+    const nextLocale = event.target.value;
+    startTransition(() => {
+      router.replace(pathname, { locale: nextLocale });
+    });
+  }
+  return (
+    <select className="lang-selector" onChange={onSelectChange} value={locale}>
+      <option value="en">🇬🇧 English</option>
+      <option value="de">🇩🇪 Deutsch</option>
+    </select>
+  );
+};
+```
+
+If you encounter any issues making it work, you can clone [the example app](https://github.com/tolgee/next-app-example) and kickstart your journey from there.
+
+## Limitations of Server Components
+
+Although in-context translation works with server components, there are some limitations compared to client components. Since the Tolgee cache on the server is separate, Tolgee can't automatically change the translation when creating a screenshot (unlike with client components, which swap the content if you've modified it in a dialog).
+
+Furthermore, if you're using the Tolgee plugin, it won't affect the server's transition to dev mode. As a result, only the client switches, leaving server components non-editable in this mode.
+
+## Conclusion and future steps
+
+Given that server components exist in a completely distinct environment, I might create a separate next.js package in the future. Yet, considering the app router is still in beta and Next.js API adjustments might occur, I opted for this article instead to explain how Tolgee can work with the server.
```

Comment:
We are company, so "we might create"

"I opted for this article instead to explain how Tolgee can work with the server." this should be stated at the beginning.